### ISSUE TITLE
some build task with `java-test-fixtures` failed in gradle 8.4, if jar task disabled

### ISSUE THREAD

**Issue by hanqyu** [gradle/gradle#27042]
### Current Behavior

When I build project, happens error like
```
* What went wrong:
Execution failed for task ':projectname:kaptGenerateStubsTestFixturesKotlin'.
> Could not resolve all files for configuration ':projectname:detachedConfiguration8'.
   > Failed to transform projectname-0.0.1-SNAPSHOT-plain.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for ClasspathEntrySnapshotTransform: /projectpath/build/libs/projectname-0.0.1-SNAPSHOT-plain.jar.
         > Check failed.
```

### Expected Behavior

Should run successfully

### Context (optional)

using spring boot3, jdk 17 

It works if I downgraded gradle to version 8.3

### Steps to Reproduce
1. clone this project https://github.com/hanqyu/gradle-27042-reproducer
2. run `./gradlew build`

### Gradle version

8.4

**Comment by ov7a** [comment]
Sorry that you're having trouble with Gradle!

We appreciate the effort that went into filing this issue, but we must ask for more information. 

As stated in our issue template, a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) is a must for us to be able to track down and fix your problem efficiently. Our available resources are severely limited, and we must be sure we are looking at the exact problem you are facing.

If we have a reproducer, we may be able also to suggest workarounds or ways to avoid the problem.

The ideal way to provide a reproducer is to leverage our [reproducer template](https://github.com/gradle/gradle-issue-reproducer). You can also use [Gradle Project Replicator](https://github.com/android/project-replicator) to reproduce the structure of your project. 

This issue will be closed after 7 days unless you provide more information.

**Comment by hanqyu** [comment]
I just updated about reproducer, to original content

>Steps to Reproduce
> 1. clone this project [hanqyu/gradle-27042-reproducer](https://github.com/hanqyu/gradle-27042-reproducer)
> 2. run ./gradlew build

and you can find the result 
https://github.com/hanqyu/gradle-27042-reproducer/actions/runs/6887183372/job/18733979742
in attached repo's action.

**Comment by cobexer** [comment]
Thank you for providing a valid report.

The issue is in the backlog of the relevant team and is prioritized by them.

**Comment by big-guy** [comment]
@hanqyu When I try to build your reproducer with 8.3, I get the same failure: https://scans.gradle.com/s/m2moa4s25meck/failure#1

Is there something else you're changing than just the Gradle version?

**Comment by big-guy** [comment]
This looks similar: https://youtrack.jetbrains.com/issue/KT-62101/IC-Execution-failed-for-ClasspathEntrySnapshotTransform-when-using-tools.jar-as-dependency

Could you try running with 8.3 and a clean build?

**Comment by hanqyu** [comment]
I tried again with 8.3, and I found the same error, you was right. 

I agree with the idea that it is not about the gradle wrapper version.


This is the how I found the error in first time.
1. I added the plugin java-test-fixtures to the module, where jar task disabled.
2. I found some error, and I changed the gradle wrapper version to 8.3

but somehow it is obviously step 2 was not the clean state. 

Is there anything I can help?
